### PR TITLE
[1LP][RFR] Fixed BootstrapSwitchSelect view

### PR DESCRIPTION
**Pull request by quarckster**
Purpose
=================
BootstrapSwitchSelect stopped to work after https://github.com/RedHatQE/widgetastic.patternfly/pull/20 merging.

{{pytest: -v -k "test_assign_two_random_events_to_control_policy" --long-running}}

### CODE REVIEW

File: widgetastic_manageiq/__init__.py [hunk 1]
```diff
@@ -674,59 +674,68 @@ def read(self):
...
 
-    def checkbox_by_id(self, id):
-        """Finds bootstrap switch by id."""
-        return BootstrapSwitch(self, id=id)
+    def switch_by_text(self, text):
+        id_attr = self._get_attr_from_text("id", text)
+        return self._bootstrap_switch(id_attr).switch
 
-    @property
+    @cached_property
```
Comment (mshriver) [inline widgetastic_manageiq/__init__.py]: Why is this cached? I see you're deleting it on fill which is good, but I'm not seeing the advantage to caching it

Comment (quarckster) [reply widgetastic_manageiq/__init__.py]: Because reading of it takes long time

File: widgetastic_manageiq/__init__.py [hunk 2]
```diff
@@ -674,59 +674,68 @@ def read(self):
...
+    def _get_attr_from_text(self, attr, text):
+        return self.browser.element(self.BS_LOCATOR.format(text)).get_attribute(attr)
 
-        Returns: str
+    @cached_property
+    def all_labels(self):
+        elements = self.browser.elements(".//div[@class='form-horizontal']")
+        labels = []
+        for el in elements:
+            labels.extend(self.browser.text(el).split(" Yes No ")[1:])
```
Comment (mfalesni) [inline widgetastic_manageiq/__init__.py]: Man, this is sooo crude :D

Comment (quarckster) [reply widgetastic_manageiq/__init__.py]: I didn't find better way to get these strings 👅

File: widgetastic_manageiq/__init__.py [hunk 3]
```diff
@@ -674,59 +674,68 @@ def read(self):
...
+    def switch_by_text(self, text):
+        id_attr = self._get_attr_from_text("id", text)
+        return self._bootstrap_switch(id_attr).switch
 
-    @property
+    @cached_property
     def selected_text(self):
-        """Only selected bootstrap switches' text descriptions."""
-        return {self._get_bs_description(bs) for bs in self.selected_checkboxes}
+        return [label for label in self.all_labels if self.switch_by_text(label).read()]
```
Comment (mfalesni) [inline widgetastic_manageiq/__init__.py]: Can you replace the read with the specific property/method? I don't want the read method's signature cluttering the logs in this case :)

**Pull request merged**